routing/http/client/client.go

```diff
@@ -79,12 +82,18 @@ func WithIdentity(identity crypto.PrivKey) Option {
 	}
 }
 
+// WithHTTPClient sets a custom HTTP Client to be used with [Client]. If you use
+// this option, [WithUserAgent] will no longer work, as it is made to work with
+// the default HTTP Client.
 func WithHTTPClient(h httpClient) Option {
 	return func(c *Client) {
 		c.httpClient = h
 	}
 }
 
+// WithUserAgent sets a custom user agent to use with the HTTP Client. This will
+// only work if you use the default HTTP Client. If you use [WithHTTPClient], you
+// should set the user agent yourself.
```

Comment:
Either this should only work if `WithHTTPClient` isn't set or it should clarify that "default HTTP Client" means `http.Client`  (with a ResponseBodyLimitedTransport transport) rather than the default one for the routing client.